.github/workflows/build-and-release.yml

```
@@ -0,0 +1,65 @@
+name: Build and Release
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go-version: [1.15, 1.16]
+        platform: [amd64, arm64]
+
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: |
+        GOOS=linux GOARCH=${{ matrix.platform }} go build -o myapp-agent ./agent/cmd/runner
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: myapp-agent-${{ matrix.platform }}
+        path: myapp-agent
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: myapp-agent-${{ matrix.platform }}
+
+    - name: Create Release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: v1.0.0
+        release_name: Release v1.0.0
+        draft: false
+        prerelease: false
+
```

from SecurityExpert
GITHUB_TOKENはGitHub Actionsのデフォルトのシークレットですが、必要最小限の権限を持つように設定することが推奨されます。特に、リリース作成に必要な権限のみを付与するようにしてください。

```suggestion
env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # 必要な権限を確認し、最小限に設定
```